### PR TITLE
support udt listening and dialing

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: convert.go [hunk 1]
```diff
@@ -56,6 +57,27 @@ func FromNetAddr(a net.Addr) (ma.Multiaddr, error) {
...
 		return ipm.Encapsulate(udpm), nil
 
+	case "udt", "udt4", "udt6":
+		ac, ok := a.(*udt.UDTAddr)
+		if !ok {
+			return nil, errIncorrectNetAddr
+		}
+
+		// Get IP Addr
+		ipm, err := FromIP(ac.UDPAddr().IP)
```
Comment (jbenet) [inline convert.go]: may want to abstract out this part common to `udp, utp, udt`. but i'm fine to merge

File: net.go [hunk 2]
```diff
@@ -232,6 +240,8 @@ func Listen(laddr ma.Multiaddr) (Listener, error) {
 	case "utp":
 		// nl, err = utp.Listen(lnet, lnaddr)
 		return nil, fmt.Errorf("utp is currently broken")
+	case "udt", "udt4", "udt6":
```
Comment (jbenet) [inline net.go]: maybe just `udt`?

File: net.go [hunk 3]
```diff
@@ -114,9 +116,15 @@ func (d *Dialer) Dial(remote ma.Multiaddr) (Conn, error) {
 		if err != nil {
 			return nil, err
 		}
+
+	case "udt", "udt4", "udt6":
```
Comment (jbenet) [inline net.go]: maybe just `udt`? not sure.

File: net.go (hunk 2 above)
Comment (whyrusleeping) [inline net.go]: i think that multiaddr might yield a 4 or a 6? i could have sworn i saw it. (could be crazy)

Comment (jbenet) [inline net.go]: ok

**Comment by jbenet** [comment]
tests fail

**Comment by jbenet** [comment]
(needs change to the script i think)

**Comment by whyrusleeping** [comment]
I'm going to kill godeps. It left out the 'udt4' directory that contains all the actual c++ udt source, and it refuses to leave it in

**Comment by jbenet** [comment]
:gun: :bomb:

**Comment by whyrusleeping** [comment]
@jbenet mind if i use gx to vendor go-udtwrapper in?

**Comment by jbenet** [comment]
Go for it. Maybe try switching all the deps?



—

**Comment by whyrusleeping** [comment]
now i'm stuck in cgo hell.

Also, gx wont be as effective as we want it to be until we switch over to 1.5 and use the `-vendor` flag. So I think we should hold off for a little bit longer on switching everything

**Comment by whyrusleeping** [comment]
i think i need to close and reopen this...

**Pull request closed without merging**